src/Aspire.Hosting.Azure/RoleDefinition.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.Azure;
+
+/// <summary>
+/// 
+/// </summary>
+/// <param name="Id"></param>
+/// <param name="Description"></param>
+
+// REVIEW: This should be part of the Azure.Provisioning APIs
+public record struct RoleDefinition(string Id, string Description);
```

Comment:
I think we need a single type for role definitions and then per resource types that have the well known ones. That would reduce the API explosion.

cc @tg-msft